docs/.vuepress/theme/components/NavVersionedLink.vue

```diff
@@ -26,10 +26,20 @@ export default {
       return ensureExt(this.item.link);
     },
     versionedLink() {
-      if (this.$page.currentVersion === this.$page.latestVersion || this.$page.DOCS_VERSION) {
-        return ensureExt(this.item.link);
+      const link = this.item.link;
+
+      const framework = this.$page.currentFramework ?
+        `/${this.$page.currentFramework}${this.$page.frameworkSuffix}` : '';
+
+      // Already have version and framework placed in the url for some reason and a `RouterLink` process it properly.
+      if (this.$page.DOCS_VERSION && (this.$page.DOCS_FRAMEWORK || this.$page.isFrameworked === false)) {
+        return ensureExt(link);
+
+      } else if (this.$page.currentVersion === this.$page.latestVersion) {
```

Comment:
Same comment as here: https://github.com/handsontable/handsontable/pull/9482#discussion_r874443296

Reply:
I'll change it after merge.

Reply:
I think that helper takes as a variable the current version and compares it with the latest version is unnecessary. If you think that something may be changed yet, please add a comment within https://github.com/handsontable/handsontable/pull/9482.